compiler/src/jdk.graal.compiler/src/jdk/graal/compiler/replacements/nodes/ArrayFillNode.java

```diff
@@ -0,0 +1,202 @@
+import jdk.graal.compiler.core.common.type.StampFactory;
+import jdk.graal.compiler.graph.NodeClass;
+import jdk.graal.compiler.lir.GenerateStub;
+import jdk.graal.compiler.nodeinfo.InputType;
+import jdk.graal.compiler.nodeinfo.NodeCycles;
+import jdk.graal.compiler.nodeinfo.NodeInfo;
+import jdk.graal.compiler.nodeinfo.NodeSize;
+import jdk.graal.compiler.nodes.FrameState;
+import jdk.graal.compiler.nodes.NamedLocationIdentity;
+import jdk.graal.compiler.nodes.StateSplit;
+import jdk.graal.compiler.nodes.ValueNode;
+import jdk.graal.compiler.nodes.memory.MemoryAccess;
+import jdk.graal.compiler.nodes.spi.NodeLIRBuilderTool;
+import jdk.vm.ci.aarch64.AArch64;
+import jdk.vm.ci.code.Architecture;
+import jdk.vm.ci.meta.JavaKind;
+
+// JaCoCo Exclude
+
+/**
+ * Fills in an array with a given constant value.
+ */
+@NodeInfo(cycles = NodeCycles.CYCLES_UNKNOWN, size = NodeSize.SIZE_UNKNOWN, allowedUsageTypes = InputType.Memory)
+public class ArrayFillNode extends MemoryKillStubIntrinsicNode
+                implements StateSplit, MemoryAccess {
+
+    public static final NodeClass<ArrayFillNode> TYPE = NodeClass.create(ArrayFillNode.class);
+    private static final ForeignCallDescriptor BOOLEAN_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Boolean);
+    private static final ForeignCallDescriptor BYTE_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Byte);
+    private static final ForeignCallDescriptor CHAR_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Char);
+    private static final ForeignCallDescriptor SHORT_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Short);
+    private static final ForeignCallDescriptor INT_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Int);
+    private static final ForeignCallDescriptor LONG_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Long);
+    private static final ForeignCallDescriptor FLOAT_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Float);
+    private static final ForeignCallDescriptor DOUBLE_ARRAY_FILL_STUB = createForeignCallDescriptor(JavaKind.Double);
+
+    public static final ForeignCallDescriptor[] STUBS = {
+                    BOOLEAN_ARRAY_FILL_STUB,
+                    BYTE_ARRAY_FILL_STUB,
+                    CHAR_ARRAY_FILL_STUB,
+                    SHORT_ARRAY_FILL_STUB,
+                    INT_ARRAY_FILL_STUB,
+                    LONG_ARRAY_FILL_STUB,
+                    FLOAT_ARRAY_FILL_STUB,
+                    DOUBLE_ARRAY_FILL_STUB,
+    };
+
+    protected JavaKind elementKind;
+
+    @Input protected ValueNode arrayBase;
+    @Input protected ValueNode offsetToFirstElement;
+    @Input protected ValueNode arrayLength;
+    @Input protected ValueNode valueToFillWith;
+
+    public ArrayFillNode(ValueNode arrayBase, ValueNode offsetToFirstElement, ValueNode arrayLength, ValueNode valueToFillWith, @ConstantNodeParameter JavaKind elementKind) {
+        super(TYPE, StampFactory.forVoid(), null, LocationIdentity.any());
```

Comment:
Please unify with the other constructor:
```suggestion
        this(array, offsetToFirstElement, arrayLength, valueToFillWith, elementKind, null);
```
Though I'm not even sure you need to pass the `runtimeCheckedCPUFeatures` everywhere.